main/src/selenium/actionWrapper.js

```diff
@@ -0,0 +1,224 @@
+    }
+    return this.action;
+  }
+  getType() {
+    return this.type;
+  }
+}
+
+class LegacyActionWrapper extends ActionWrapper {
+  constructor(driverForActions, renderer) {
+    super(driverForActions);
+    this.renderer = renderer;
+    this.actionList = [];
+    this.BREAK_MOVE_LENGTH = 10;
+    this.pressPauseDuration = 20;
+    this.zoomPauseDuration = 500;
+  }
+  async perform() {
+    for (let index = 0; index < this.actionList.length; index += 1) {
+      const action = this.actionList[index];
+      if (action.getType() === typePause) {
+        await sleep(action.getPauseDuration());
+      } else if (action.getType() === typeActionLegacySequence) {
+        await action.getActionLegacySequence().perform();
+      }
+    }
+  }
+  moveToStartPoint() {
+    this.actionList.push(new LegacyActionExtended(
+      typeActionLegacySequence,
+      new LegacyActionSequence(this.driverForActions).mouseMove(this.map),
+    ));
+    return this;
+  }
+  pause(duration) {
+    this.actionList.push(new LegacyActionExtended(
+      typePause,
+      duration,
+    ));
+    return this;
+  }
+  drag(duration, x, y) {
+    this._press();
+    const action = new LegacyActionSequence(this.driverForActions);
+    for (let index = 0; index < this.BREAK_MOVE_LENGTH; index += 1) {
+      action.mouseMove({
+        x: Math.floor(x / this.BREAK_MOVE_LENGTH),
+        y: Math.floor(y / this.BREAK_MOVE_LENGTH),
+      });
+    }
+    action.mouseMove({ x: x % this.BREAK_MOVE_LENGTH, y: y % this.BREAK_MOVE_LENGTH });
+    this.actionList.push(new LegacyActionExtended(
+      typeActionLegacySequence,
+      action,
+    ));
+    this.actionList.push(new LegacyActionExtended(
+      typePause,
+      this.pressPauseDuration,
+    ));
+    this._release();
+    this.actionList.push(new LegacyActionExtended(
+      typeActionLegacySequence,
+      new LegacyActionSequence(this.driverForActions).mouseMove({ x: -x, y: -y }),
+    ));
+    this.actionList.push(new LegacyActionExtended(
+      typePause,
+      duration,
+    ));
+    return this;
+  }
+  _press() {
+    this.actionList.push(new LegacyActionExtended(
+      typeActionLegacySequence,
+      new LegacyActionSequence(this.driverForActions).mouseDown(),
+    ));
+    this.actionList.push(new LegacyActionExtended(
+      typePause,
+      this.pressPauseDuration,
+    ));
+  }
+  _release() {
+    this.actionList.push(new LegacyActionExtended(
+      typeActionLegacySequence,
+      new LegacyActionSequence(this.driverForActions).mouseUp(),
+    ));
+    this.actionList.push(new LegacyActionExtended(
+      typePause,
+      this.pressPauseDuration,
+    ));
+  }
+  doubleClick() {
+    this.moveToStartPoint();
+    if (this.renderer === 'openlayers') {
+      this.actionList.push(new LegacyActionExtended(
+        typeActionLegacySequence,
+        new LegacyActionSequence(this.driverForActions).click().click(),
```

Comment:
It would be good to explain why the different actions are required for the two renderers.